valhalla/midgard/encoded.h

```diff
@@ -95,13 +97,16 @@ template <class container_t> std::string encode(const container_t& points) {
     output.push_back(static_cast<char>(number));
   };
 
+  // Set precision - only allows 5 or 6. If any other value we go back to the default (6).
+  int prec = (precision == 5) ? 1e5 : 1e6;
```

Comment:
we could just allow the caller to pass in the precision instead of the nubmer of decimal places and avoid this `if` and extra variable. its more flexible that way as well.

Reply:
Agreed.  If someone passes in `polyline4` here, they'll get a polyline6, which would be [a surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

Reply:
done